lib/core/service/file_service.dart

```diff
@@ -58,9 +58,16 @@ class FileService {
 
   static Future<Directory> get downloadsDirectory async {
     if (Platform.isAndroid) {
-      return (await getExternalStorageDirectories(
-              type: StorageDirectory.downloads))!
-          .first;
+      Directory? directory;
+      directory = Directory('/storage/emulated/0/Download');
+      // Try the default global folder, if it exists
+      if (!await directory.exists()) {
+        directory = (await getExternalStorageDirectories(
+          type: StorageDirectory.downloads,
+        ))!
+            .first;
+      }
+      return directory;
     } else if (Platform.isIOS) {
       return getApplicationDocumentsDirectory();
```

Comment:
Is there a reason why you chose not to use the `getExternalStorageDirectories(type: StorageDirectory.downloads))` for platforms other than android?

Reply:
As I don't have any iOS devices, I couldn't test if it works, so I preferred to leave everything else untouched, I only changed the parts in the "if (android)" guard.
Also, if my memory doesn't betray me, as I was looking for the proper solution, i saw that for iOS the proper solution is in fact the getApplicationDocumentsDirectory();, as implemented.
I can try to test on an IOS emulator and see the differences.
Later edit: getExternalStorageDirectories seems only to work on Android. What I could use instead is getApplicationDocumentsDirectory for all platforms, as it should work. Again, this was a rabbit hole, so many versions so many solutions.